internal/controller/watch/predicates.go

```diff
@@ -58,11 +65,25 @@ func SelectNamespacesPredicate(namespaces []string) predicate.Funcs {
 	})
 }
 
+// ReconcileAgainPredicate reconciles on updates when the skip reconcile
+// annotation is removed or changed
+func ReconcileAgainPredicate[T metav1.Object]() predicate.TypedPredicate[T] {
```

Comment:
```suggestion
func SkipAnnotationRemovedPredicate[T metav1.Object]() predicate.TypedPredicate[T] {
```

The name `ReconcileAgainPredicate` is too general, all predicates decide whether a reconcile happens "again"